kagen/context.cpp

```diff
@@ -342,6 +342,12 @@ PGeneratorConfig CreateConfigFromString(const std::string& options_str, PGenerat
     config.coordinates = get_bool_or_default("coordinates");
     config.permute     = get_bool_or_default("permute");
 
+    // edge weight generation
+    const std::string edge_weight_generator_name = get_string_or_default("edgeweight-generator", "none");
+    config.edge_weights.generator_type = GetEdgeWeightGeneratorTypeMap()[edge_weight_generator_name];
```

Comment:
add error checking similar to the other enum options so that a misspelled generator type is not silently ignored